Урок 5. Прямолинейное движение

dmitriyaf74 opened the pull request.

dmitriyaf74 committed:
add straight files

dmitriyaf74 committed:
Прямлонейное движение

71andy commented:
Нужно привести код в соответствие с PEP8, почистить все ошибки, которые линтер flake8 выдает. 
flake8 можно запускать локально, перед коммитами. Также можно и хорошо настроить форматтер в редакторе, чтобы вся эта работа по форматированию выполнялась автоматически

dmitriyaf74 committed:
add pep8

vakhet commented on space_invaders/engine/commands/change_straight_line_position.py:
```diff
@@ -0,0 +1,22 @@
+            raise exceptions.EObjectNotMoveableError
+        if None in [self._obj.current_position, self._obj.position_correction]:
+            raise exceptions.ENoneStraightLinePositionError
+        if None in self._obj.current_position + self._obj.position_correction:
```
в чем смысл такой проверки? как может результат сложения оказаться массивом с None ??

dmitriyaf74 replied:
Это проверка входных данных

dmitriyaf74 replied:
Спасибо, это не результат сложения, а проверка входных значений, т.е. сложение 2х кортежей

dmitriyaf74 replied:
Конечно, логичнее проверять без сложения, исправлю

vakhet commented on space_invaders/engine/exceptions.py:
```diff
@@ -0,0 +1,6 @@
+class ENoneStraightLinePositionError(Exception):
+    """Исклюение при получении не определено значение координат"""
```
опечатка, здесь и ниже

dmitriyaf74 replied:
Спасибо

vakhet commented on space_invaders/engine/interfaces/__init__.py:
```diff
@@ -0,0 +1,8 @@
+
+__all__ = (
+    "Command",
+    "StraightLinePositionController"
```
запятая в конце строки

dmitriyaf74 replied:
для чего? 
На всякий случай поставил))

vakhet replied:
чтобы при добавлении нового значения в списке не изменялись существующие строки

vakhet commented on space_invaders/engine/interfaces/straight_line_position_controller.py:
```diff
@@ -0,0 +1,30 @@
+
+
+class StraightLinePositionController(ABC):
+    """Интерфейс для перемещаемого объекта."""
```
эта абстракция входит в область значений Movable, можно смело переиспользовать ее

dmitriyaf74 replied:
Не понял что ты хочешь сказать, Вадим, можешь объяснить?

vakhet replied:
Movable - сущности которые перемещаются в пространстве,  они уже вроде бы реализованы... возможно я не прав, кто как считает?

71andy replied:
Я не понял, где они реализованы? У нас же каждое задание - это отдельный PR в фактически пустой реп.

vakhet commented on space_invaders/tests/commands/test_change_straight_line_position.py:
```diff
@@ -0,0 +1,86 @@
+import pytest
+from space_invaders.engine import exceptions
+from space_invaders.engine.interfaces import StraightLinePositionController
+from space_invaders.engine.commands.change_straight_line_position \
```
```suggestion
from space_invaders.engine.commands import ChangeStraitLinePosition
```

dmitriyaf74 replied:
Спасибо

vakhet commented on space_invaders/engine/commands/change_straight_line_position.py:
```diff
@@ -0,0 +1,22 @@
+import operator
+
+
+class ChangeStraightLinePosition(Command):
```
предлагаю переименовать на MoveStraightLine

dmitriyaf74 replied:
Переименую просто в Move

dmitriyaf74 committed:
Работа с замечаниями

Переименовал StraightLinePosition в Move
удалил свойство moveable интерфейса
удалил типизацию свойств интерфейса

71andy commented on space_invaders/engine/interfaces/straight_line_position_controller.py:
```diff
@@ -0,0 +1,20 @@
+    @property
+    @abstractmethod
+    def velocity(self):
+        """Возвращает координаты после перемещения"""
```
Комментарий к функции не понятен. Должна возвращаться мгновенная скорость

dmitriyaf74 replied:
исправил

dmitriyaf74 committed:
эстетические корректировки

71andy commented on space_invaders/engine/commands/change_straight_line_position.py:
```diff
@@ -0,0 +1,22 @@
+        if None in [self._obj.position, self._obj.velocity]:
+            raise exceptions.ENoneMoveError
+
+        if None in self._obj.position + self._obj.velocity:
```
проверку на None правильнее делать 
`if var is None:`

dmitriyaf74 replied:
исправил

dmitriyaf74 committed:
Замечания 2

исправлен комментарий к velocity
исправлена проверка на None

71andy commented on space_invaders/engine/exceptions.py:
```diff
@@ -0,0 +1,2 @@
+class ENoneMoveError(Exception):
+    """Исключение при получении не определено значение координат"""
```
Мне такое наименование исключения не понятно. При чтении кода я понимаю, что при перемещении вдруг получились координаты `None`. Сразу возникает вопросы, а почему они получились `None`? К тому же это исключение используется при разных ошибках (при ошибке чтения скорости, чтения текущих координат и установке координат). Лучше назвать `MoveError, ObjectMoveError` или еще как-то, что однозначно обобщит все ошибки перемещения объекта. Либо вводить несклько исключений, который будут конкретизировать причину ошибки: `GetPositionError, SetPositionError, GetVelocityError`. Но сейчас я не могу понять, понадобится ли эта детализация в дальнейшем.

dmitriyaf74 committed:
Замечания 71andy

Изменен тип исключений с ENoneMoveError на EGetPositionError,EGetVelocityError

dmitriyaf74 committed:
Merge branch 'main' into dfilippov_home_work_2

dmitriyaf74 committed:
update code for new requirements flake8

changed some RU-comments to ENG-comments
correcting code requirements

dmitriyaf74 committed:
update code for new requirements flake8 attempt 2

dmitriyaf74 committed:
Update test_change_straight_line_position.py
